storage/src/mmr/verification.rs

```diff
@@ -324,9 +333,26 @@ impl<D: Digest> Proof<D> {
         start_element_pos: u64,
         end_element_pos: u64,
     ) -> Result<Proof<D>, Error> {
+        Self::historical_range_proof(mmr, mmr.size(), start_element_pos, end_element_pos).await
+    }
+
+    /// Analagous to range_proof but for a previous database state.
+    /// Specifically, the state when the MMR had `size` elements.
+    pub async fn historical_range_proof<S: Storage<D>>(
+        mmr: &S,
+        size: u64,
+        start_element_pos: u64,
+        end_element_pos: u64,
+    ) -> Result<Proof<D>, Error> {
+        if size > mmr.size() {
+            return Err(Error::HistoricalSizeTooLarge(size, mmr.size()));
+        }
+        if size <= start_element_pos {
+            return Err(Error::HistoricalSizeTooSmall(size, start_element_pos));
```

Comment:
I wonder if we should be using `ElementPruned` for this?

Reply:
Personally, I like this since it makes it immediately apparent what the fault is. Either way is fine by me.